pkg/ebpf/tracee.go

```diff
@@ -267,6 +269,12 @@ func New(cfg Config) (*Tracee, error) {
 	}
 	t.containers = c
 
+	t.kernelSymbols, err = helpers.NewKernelSymbolsMap()
+	if err != nil {
+		t.Close()
+		return nil, fmt.Errorf("error creating symbols map: %v", err)
+	}
```

Comment:
I think we should only initialize this map if there's an event that actually uses it.
This way, even if there is no access to kallsysms for some reason, we will not totally fail tracee.

Reply:
I can argue with it, because using the symbols is not only relevant for the eBPF programs.
For example, if the kconfig file is not available in the distro we want to use it to determine the kconfigs used in compilation. This is done in the usermode, though you can argue that in that case we can request the data sepcifically.

If you want me to implement the check if some program use it, it will also add the overhead of changing the events definitions interface to add dependencies (we can add the use of the kconfig map as well). I am not rejecting the idea, just asking if this is what you want me to implement.

Reply:
More argument to do it is to enable more flexibility of eBPF programs loading. If in the future we would like to load an BPF program on the fly, we would like to have all information needed beforehand (or to load it on the fly). It doesn't mean that we have to load it beforehand, but it means that if you want us not to load it for removing some capabilities for example it will make it difficult to support dynamic loading, so I don't think it will help from the security aspect too much.